File: lib/Models/CampaignSettingsDTO.cpp


#include "CampaignSettingsDTO.h"

using namespace Tiny;

CampaignSettingsDTO::CampaignSettingsDTO()
{
	countryRegion = long(0);
	shopName = std::string();
	showInContext = bool(false);
	showInPremium = bool(false);
	useOpenStat = bool(false);
	localRegion = CampaignSettingsLocalRegionDTO();
}

CampaignSettingsDTO::CampaignSettingsDTO(std::string jsonString)
{
	this->fromJson(jsonString);
}

CampaignSettingsDTO::~CampaignSettingsDTO()
{

}

void
CampaignSettingsDTO::fromJson(std::string jsonObj)
{
    bourne::json object = bourne::json::parse(jsonObj);

    const char *countryRegionKey = "countryRegion";

    if(object.has_key(countryRegionKey))
    {
        bourne::json value = object[countryRegionKey];



        jsonToValue(&countryRegion, value, "long");


    }

    const char *shopNameKey = "shopName";

    if(object.has_key(shopNameKey))
    {
        bourne::json value = object[shopNameKey];



        jsonToValue(&shopName, value, "std::string");


    }

    const char *showInContextKey = "showInContext";

    if(object.has_key(showInContextKey))
    {
        bourne::json value = object[showInContextKey];



        jsonToValue(&showInContext, value, "bool");


    }

    const char *showInPremiumKey = "showInPremium";

    if(object.has_key(showInPremiumKey))
    {
        bourne::json value = object[showInPremiumKey];



        jsonToValue(&showInPremium, value, "bool");


    }

    const char *useOpenStatKey = "useOpenStat";

    if(object.has_key(useOpenStatKey))
    {
        bourne::json value = object[useOpenStatKey];



        jsonToValue(&useOpenStat, value, "bool");


    }

    const char *localRegionKey = "localRegion";

    if(object.has_key(localRegionKey))
    {
        bourne::json value = object[localRegionKey];




        CampaignSettingsLocalRegionDTO* obj = &localRegion;
		obj->fromJson(value.dump());

    }


}

bourne::json
CampaignSettingsDTO::toJson()
{
    bourne::json object = bourne::json::object();





    object["countryRegion"] = getCountryRegion();






    object["shopName"] = getShopName();






    object["showInContext"] = isShowInContext();






    object["showInPremium"] = isShowInPremium();






    object["useOpenStat"] = isUseOpenStat();







	object["localRegion"] = getLocalRegion().toJson();


    return object;

}

long
CampaignSettingsDTO::getCountryRegion()
{
	return countryRegion;
}

void
CampaignSettingsDTO::setCountryRegion(long  countryRegion)
{
	this->countryRegion = countryRegion;
}

std::string
CampaignSettingsDTO::getShopName()
{
	return shopName;
}

void
CampaignSettingsDTO::setShopName(std::string  shopName)
{
	this->shopName = shopName;
}

bool
CampaignSettingsDTO::isShowInContext()
{
	return showInContext;
}

void
CampaignSettingsDTO::setShowInContext(bool  showInContext)
{
	this->showInContext = showInContext;
}

bool
CampaignSettingsDTO::isShowInPremium()
{
	return showInPremium;
}

void
CampaignSettingsDTO::setShowInPremium(bool  showInPremium)
{
	this->showInPremium = showInPremium;
}

bool
CampaignSettingsDTO::isUseOpenStat()
{
	return useOpenStat;
}

void
CampaignSettingsDTO::setUseOpenStat(bool  useOpenStat)
{
	this->useOpenStat = useOpenStat;
}

CampaignSettingsLocalRegionDTO
CampaignSettingsDTO::getLocalRegion()
{
	return localRegion;
}

void
CampaignSettingsDTO::setLocalRegion(CampaignSettingsLocalRegionDTO  localRegion)
{
	this->localRegion = localRegion;
}

File: lib/app/Models/CampaignSettingsDTO.php
<?php
/**
 * CampaignSettingsDTO
 */
namespace app\Models;

/**
 * CampaignSettingsDTO
 * @description Настройки магазина.
 */
class CampaignSettingsDTO {

    /** @var int $countryRegion Идентификатор региона, в котором находится магазин.*/
    public $countryRegion = 0;

    /** @var string $shopName Наименование магазина на Яндекс Маркете. Если наименование отсутствует, значение параметра выводится — &#x60;null&#x60;.*/
    public $shopName = "";

    /** @var bool $showInContext Признак размещения магазина на сайтах партнеров Яндекс Дистрибуции. Возможные значения: * &#x60;false&#x60; — магазин не размещен на сайтах партнеров Яндекс Дистрибуции. * &#x60;true&#x60; — магазин размещен на сайтах партнеров Яндекс Дистрибуции.*/
    public $showInContext = false;

    /** @var bool $showInPremium Признак показа предложений магазина в рекламном блоке над результатами поиска (cпецразмещение). Возможные значения: * &#x60;false&#x60; — предложения не показываются в блоке cпецразмещения. * &#x60;true&#x60; — предложения показываются в блоке cпецразмещения.*/
    public $showInPremium = false;

    /** @var bool $useOpenStat Признак использования внешней интернет-статистики. Возможные значения: * &#x60;false&#x60; — внешняя интернет-статистика не используется. * &#x60;true&#x60; — внешняя интернет-статистика используется.*/
    public $useOpenStat = false;

    /** @var \app\Models\CampaignSettingsLocalRegionDTO $localRegion */
    public $localRegion;

}
